src/Database/Query.php

```diff
@@ -522,41 +522,37 @@ public function build(string $type)
 	{
 		$sql = $this->database->sql();
 
-		switch ($type) {
-			case 'select':
-				return $sql->select([
-					'table'    => $this->table,
-					'columns'  => $this->select,
-					'join'     => $this->join,
-					'distinct' => $this->distinct,
-					'where'    => $this->where,
-					'group'    => $this->group,
-					'having'   => $this->having,
-					'order'    => $this->order,
-					'offset'   => $this->offset,
-					'limit'    => $this->limit,
-					'bindings' => $this->bindings
-				]);
-			case 'update':
-				return $sql->update([
-					'table'    => $this->table,
-					'where'    => $this->where,
-					'values'   => $this->values,
-					'bindings' => $this->bindings
-				]);
-			case 'insert':
-				return $sql->insert([
-					'table'    => $this->table,
-					'values'   => $this->values,
-					'bindings' => $this->bindings
-				]);
-			case 'delete':
-				return $sql->delete([
-					'table'    => $this->table,
-					'where'    => $this->where,
-					'bindings' => $this->bindings
-				]);
-		}
+		return match ($type) {
+			'select' => $sql->select([
+				'table'    => $this->table,
+				'columns'  => $this->select,
+				'join'     => $this->join,
+				'distinct' => $this->distinct,
+				'where'    => $this->where,
+				'group'    => $this->group,
+				'having'   => $this->having,
+				'order'    => $this->order,
+				'offset'   => $this->offset,
+				'limit'    => $this->limit,
+				'bindings' => $this->bindings
+			]),
+			'update' => $sql->update([
+				'table'    => $this->table,
+				'where'    => $this->where,
+				'values'   => $this->values,
+				'bindings' => $this->bindings
+			]),
+			'insert' => $sql->insert([
+				'table'    => $this->table,
+				'values'   => $this->values,
+				'bindings' => $this->bindings
+			]),
+			'delete' => $sql->delete([
+				'table'    => $this->table,
+				'where'    => $this->where,
+				'bindings' => $this->bindings
+			])
+		};
```

Comment:
Just for a reminder. FIY, will throw `UnhandledMatchError` when not matched, instead of `null`.  Should we do something?

Reply:
It seems so. For 3.8, we can add todo to throw exception.

Reply:
Since this will be for 3.8 anyways, maybe let's include it directly here. Just need to make sure to write down the breaking change in the notes.

Reply:
Oh, right 😄